spec/models/invoice_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe Invoice, type: :model do
+  describe 'relationships' do
+    it { should belong_to(:customer) }
+    it { should have_many(:invoice_items) }
+    it { should have_many(:transactions) }
+    it { should have_many(:items).through(:invoice_items) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:status) }
```

Comment:
Add validation test for numericality: `it { should validate_numericality_of(:status) }`